containers/src/Kokkos_Vector.hpp

```diff
@@ -196,10 +196,16 @@ class vector : public DualView<Scalar*, LayoutLeft, Arg1Type> {
 
   iterator begin() const { return DV::h_view.data(); }
 
+  const_iterator cbegin() const { return DV::h_view.data(); }
+
   iterator end() const {
     return _size > 0 ? DV::h_view.data() + _size : DV::h_view.data();
   }
 
+  const_iterator cend() const {
```

Comment:
I'll approve this because it does the same thing as `end()`, but why the `size_` check?  It is valid to add `0` to a valid pointer, even if that pointer has the null pointer value.

Reply:
You're correct! Adding 0 to a nullptr is a valid operation [(link)](https://stackoverflow.com/questions/59409034/is-it-allowed-to-add-a-zero-to-a-null-pointer#:~:text=Yes%2C%20you%20can%20add%20zero,Quoting%20Additive%20operators%20%5Bexpr.)! So, I can update this PR and remove the `_size` check if people agree !

Reply:
Not in this PR.

Reply:
Ok, let's not remove the `_size` check in this PR.